HelpSource/Classes/Object.schelp

```diff
@@ -200,6 +200,113 @@ A String containing the resulting file's path.
 
 subsection::Equality and Identity
 
+Equally asks if the value of two objects is equal.
+Identity asks if they are the same object.
+There are a few classes of object where only one instance exists and is shared between all uses (a singleton).
+These objects include Symbols and numbers.
+
+Equality, tested for with code::==:: or code::!=:: is a relation between objects.
+So code::a == b:: is true when the object in the variable a and the object in b are equivalent in some essential aspect.
+Two strings, for example, are equal when they have the same characters in the same order.
+Identity, by contrast, is a relation that every object has to itself. So code::a === b:: is true only when the the variables a and b both contain the same object.
+
+code::
+// Strings, written with double quotes, are the same from the perspective of equality.
```

Comment:
I'd remove the first comment explaining what a string is. If you use `link::Classes/String` you shouldn't need to do this. The second line could go outside the code block.

Reply:
These part is the reuse of the text writtren by @telephon:
https://github.com/supercollider/supercollider/pull/6385#discussion_r1665821000
```
Equality, tested for with code::==:: or code::!=:: is a relation between objects.
So code::a == b:: is true when the object in the variable a and the object in b are equivalent in some essential aspect.
Two strings, for example, are equal when they have the same characters in the same order.
Identity, by contrast, is a relation that every object has to itself. So code::a === b:: is true only when the the variables a and b both contain the same object.

code::
// Strings, written with double quotes, are the same from the perspective of equality.
// However, they are always different from the perspective of identity:
"we are all the same" == "we are all the same" // true
"we are all the same" === "we are all the same" // false: same, but not identical

// Symbols, written with single quotes or following \ without spaces, are special:
// There is only one of each.
// Two representations of the same symbol are identical:
'we are all the same' == 'we are all the same' // true
'we are all the same' === 'we are all the same' // true
::
```
As @telephon and @JordanHendersonMusic suggest different opinions, I will reconstruct the part as follows:
```
Equally asks if the value of two objects is equal.
Identity asks if they are the same object.
There are a few classes of object where only one instance exists and is shared between all uses (a singleton).
These objects include symbols (the instances of link::Classes/Symbol::) and numbers (the instances of the subclasses of link::Classes/SimpleNumber::: link::Classes/Integer:: and link::Classes/Float::).

Equality, tested for with code::==:: or code::!=:: is a relation between objects.
So code::a == b:: is true when the object in the variable a and the object in b are equivalent.
Identity, by contrast, is a relation that every object has to itself. So code::a === b:: is true only when the the variables a and b both contain the same object.

table::
## strong::Examples:::

Two link::Classes/String:: instances, for example, are equal when they have the same characters in the same order.
However, they are always different from the perspective of identity:

code::
"we are all the same" == "we are all the same" // true
"we are all the same" === "we are all the same" // false: same, but not identical
::

The instances of the class link::Classes/Symbol::, written with single quotes or following \ without spaces, are special. Since there is only one of each, two representations of the same symbol are identical:
code::
'we are all the same' == 'we are all the same' // true
'we are all the same' === 'we are all the same' // true
::
::
```
![Screenshot 2024-07-07 at 09 30 45](https://github.com/supercollider/supercollider/assets/416281/60355b44-b88e-4bad-8734-ef1b70ab0e98)

What do you think?

Reply:
You could also add a number example here with some comments.
```
1.0 == 1.0
1.0 === 1.0
1.0 == 1
1.0 =!= 1
```

Reply:
How about this?
```
Two instances of the subclasses of link::Classes/SimpleNumber:: are the same in terms of equality if they have the same value, while they are different in terms of identity if they do not belong to the same classes (in other words, their output data types are different) even though they have the same values:

code::
1.0 == 1.0 // true: the values are equal.
1.0 === 1.0 // true: the values and the classes are the same (Float).
1.0 == 1 // true: the values are equal.
1.0 === 1 // false: the same in terms of values, but the different classes
::
```
![Screenshot 2024-07-11 at 11 39 46](https://github.com/supercollider/supercollider/assets/416281/6e508bb0-5662-434b-8324-50d02541a7c2)

Reply:
> Looks good to me.
> 
> One reason why I suggested a different version was that two objects can never be identical. This makes it a little hard to describe. What is meant is that the value of two variables are the same or not, right?

This

Reply:
> Looks good to me.
> 
> One reason why I suggested a different version was that two objects can never be identical. This makes it a little hard to describe. What is meant is that the value of two variables are the same or not, right?

I think this should be answered by @JordanHendersonMusic:

@telephon wrote:
>Equality, tested for with code::==:: or code::!=:: is a relation between objects.
So code::a == b:: is true when the object in the variable a and the object in b are equivalent in some essential aspect.
Contributor

@JordanHendersonMusic suggested:
https://github.com/supercollider/supercollider/pull/6385#discussion_r1667345989
>>in some essential aspect.
>You don't need this clause, just makes things more confusing.

I think 'in some essential aspect' can be helpful but also can make things somewhat confusing. 

I would write as follows:
```
So code::a == b:: is true if the objects referenced by the variables code::a:: and code::b:: contain the same things, such as values, data or anything else.
```

Reply:
>How about this?
...
code::
1.0 == 1.0 // true: the values are equal.
1.0 === 1.0 // true: the values and the classes are the same (Float).
1.0 == 1 // true: the values are equal.
1.0 === 1 // false: the same in terms of values, but the different classes
::

This is now a little confusing for the identity ones, perhaps you could say "there is only a single instance of the value '1.0', therefore they are identical (the same object)", and some thing similar for the other identity about being different objects.

Reply:
> So code::a == b:: is true if the objects referenced by the code::a:: and code::b:: variables contain the same things, such as values or data.

This isn't clear because you've just introduced values and data as separate things when they are the same.

I wouldn't talk about objects when discussing equality, keep that concept for identity.